sycl/test/basic_tests/scalar_vec_access.cpp

```diff
@@ -0,0 +1,60 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out | FileCheck %s
+// RUN: %CPU_RUN_PLACEHOLDER %t.out %CPU_CHECK_PLACEHOLDER
+// RUN: %GPU_RUN_PLACEHOLDER %t.out %GPU_CHECK_PLACEHOLDER
+// RUN: %ACC_RUN_PLACEHOLDER %t.out %ACC_CHECK_PLACEHOLDER
+//==------- scalar_vec_access.cpp - SYCL scalar access to vec test ---------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// CHECK: C {5, 5, 5, 5}
+
+#include <CL/sycl.hpp>
+#include <iostream>
+
+typedef float float4_t __attribute__((ext_vector_type(4)));
+
+int main() {
+  // Test that it is possible to get a reference to single element of the vector
+  // type. This behavior could possibly change in the future, this test is
+  // necessary to track that.
+  float4_t my_float4 = {0.0, 1.0, 2.0, 3.0};
+  float f[4];
+  for (int i = 0; i < 4; ++i) {
+    f[i] = reinterpret_cast<float *>(&my_float4)[i];
+    assert(f[i] == i);
+  }
+
+  // Test to check access to single element of the sycl::vec type.
```

Comment:
We already had tests for that (e.g. swizzle_op.cpp). As far as I can see, this tests that "ambiguity in overload resolution" problem is fixed. A much simpler test might be sufficient for that:
```c++
cl::sycl::float4 a = {1.0, 2.0, 3.0, 4.0};
std::cout  << a.x();
```

Might be worth also adding a test for CORC-5725. E.g. calling `sycl::native::recip`.

Reply:
Thanks! Fixed. Added test for sycl::native:;recip.